src/Compilers/CSharp/Portable/Symbols/MethodSymbol.cs

```diff
@@ -967,45 +968,105 @@ protected static UnmanagedCallersOnlyAttributeData DecodeUnmanagedCallersOnlyAtt
             {
                 foreach (var (key, value) in attribute.CommonNamedArguments)
                 {
-                    if (!UnmanagedCallersOnlyAttributeData.IsCallConvsTypedConstant(key, in value))
-                    {
-                        continue;
-                    }
+                    var namedArgumentDecoded = TryDecodeUnmanagedCallersOnlyCallConvsProperty(key, value, location, diagnostics);
 
-                    if (callingConventionTypes != null)
+                    if (namedArgumentDecoded.IsCallConvs)
                     {
-                        isValid = false;
+                        isValid = isValid && namedArgumentDecoded.IsValid;
```

Comment:
>isValid [](start = 34, length = 7)

Since we are interested only in the last value, why are we paying attention to validity of other values?

Reply:
> Since we are interested only in the last value

If we're interpreting source, I don't believe it's true that we're only interested in the last thing. I use the set of the last convention as the set for simplicity, but the data in the UnmanagedCallersOnlyAttribute itself is definitively bad as you can't specify it more than once. I can likely remove this step from the IL decoding version though.

---
In reply to: [493019870](https://github.com/dotnet/roslyn/pull/47950#discussion_r493019870) [](ancestors = 493019870)

Reply:
I'm actually going to remove the `IsValid` tracking entirely. We only use it for use-site diagnostics on imported method symbols, and I don't think it's worth the amount of effort I'm putting in to track it. If we get feedback from users about wanting better error messages from bad metadata, we can reconsider.

---
In reply to: [493882174](https://github.com/dotnet/roslyn/pull/47950#discussion_r493882174) [](ancestors = 493882174,493019870)

Reply:
> If we're interpreting source, I don't believe it's true that we're only interested in the last thing.

We should be consistent even when we are recovering from errors. We recognize properties on Obsolete attribute, let's match that behavior in terms which value to use as a result of decoding in presence of duplicate named value.

---
In reply to: [493894825](https://github.com/dotnet/roslyn/pull/47950#discussion_r493894825) [](ancestors = 493894825,493882174,493019870)